cmd/web-handlers.go

```diff
@@ -667,9 +667,11 @@ next:
 	for _, objectName := range args.Objects {
 		// If not a directory, remove the object.
 		if !HasSuffix(objectName, SlashSeparator) && objectName != "" {
-			// Check for permissions only in the case of
-			// non-anonymous login. For anonymous login, policy has already
-			// been checked.
+			// Get the retention/lock info to check if it is enabled or not
+			info, _ := globalBucketObjectLockConfig.Get(args.BucketName)
```

Comment:
here the pattern should be like this 
```
info, ok := globalBucketObjectLockConfig.Get(args.BucketName)
if ok && info.Mode != "" {
    bucketObjectLockEnabled = true
}
```

Reply:
Done
Testing now one more time,  just to make sure everything works